src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -1,244 +1,20 @@
-        final File refFile;
-        final String[] args;
-        if (reference == null) {
-            refFile = null;
-            args = new String[]{
-                    "--input", originalFile.getAbsolutePath(),
-                    "--output", outFile.getAbsolutePath(),
-            };
-        } else {
-            refFile = new File(TEST_DATA_DIR, reference);
-            args = new String[]{
-                    "--input", originalFile.getAbsolutePath(),
-                    "--output", outFile.getAbsolutePath(),
-                    "-R", refFile.getAbsolutePath()
-            };
-        }
-        try (ReadsDataSource ds = reference==null ? new ReadsDataSource(originalFile.toPath()) :
-                new ReadsDataSource(originalFile.toPath(), SamReaderFactory.make().referenceSequence(refFile.toPath()))){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.queryname);
-        }
-
-        runCommandLine(args);
-        SamAssertionUtils.assertSamsEqual(outFile, originalFile, refFile);
-    }
-
-    @Test( groups = "spark")
-    public void testNameSorted() throws Exception {
-        final File inBam = new File(getTestDataDir(), "print_reads.bam");
-        try (ReadsDataSource ds = new ReadsDataSource(inBam.toPath())){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.queryname);
-        }
-        final File outBam = GATKBaseTest.createTempFile("print_reads", ".bam");
-        ArgumentsBuilder args = new ArgumentsBuilder();
-        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
-        args.add(inBam.getCanonicalPath());
-        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
-        args.add(outBam.getCanonicalPath());
-
-        this.runCommandLine(args.getArgsArray());
-
-        SamAssertionUtils.assertSamsEqual(outBam, inBam);
-    }
-
-    @Test( groups = "spark")
-    public void testUnSorted() throws Exception {
-        final File inBam = new File(getTestDataDir(), "print_reads.unsorted.bam");
-        try (ReadsDataSource ds = new ReadsDataSource(inBam.toPath())){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.unsorted);
-        }
-        final File outBam = GATKBaseTest.createTempFile("print_reads", ".bam");
-        ArgumentsBuilder args = new ArgumentsBuilder();
-        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
-        args.add(inBam.getCanonicalPath());
-        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
-        args.add(outBam.getCanonicalPath());
-
-        this.runCommandLine(args.getArgsArray());
+import java.io.File;
 
-        SamAssertionUtils.assertSamsEqual(outBam, inBam);
-    }
+public final class PrintReadsSparkIntegrationTest extends AbstractPrintReadsIntegrationTest {
 
-    @Test( groups = "spark")
+    @Test()
     public void testUnSortedRoundTrip() throws Exception {
```

Comment:
This is the only test that is Spark-specific. It fails on PrintReads (non-Spark) since that tool does not allow reads to be written in an order that is inconsistent with the header (the header says coordinate sorted, but the file is actually queryname sorted).

Also (contrary to the comment in this test) PrintReadSpark does not sort the file, so it writes an invalid BAM. This seems to be wrong.

Reply:
Right, so the point with this test as I added it was to enforce that printReadsSpark was no longer spuriously sorting the output reads (at runtime cost and some pain if you were dealing with invalid input files in the first place). I'm not sure its the place of PrintReads to go about sorting your bam output for you, especially if your purpose is to extract reads with some tag or another. Perhaps this test would make more sense if we were looking at queryname sorted bams, where there exists a valid alternative.